supernode/server/preheat_bridge.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/dragonflyoss/Dragonfly/apis/types"
+	"github.com/dragonflyoss/Dragonfly/pkg/errortypes"
+	"github.com/go-openapi/strfmt"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+)
+
+// ---------------------------------------------------------------------------
+// handlers of preheat http apis
+
+func (s *Server) createPreheatTask(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	request := &types.PreheatCreateRequest{}
+	if err := parseRequest(req.Body, request, request.Validate); err != nil {
+		return err
+	}
+	preheatID, err := s.PreheatMgr.Create(ctx, request)
+	if err != nil {
+		return err
+	}
+	resp := types.PreheatCreateResponse{ID: preheatID}
+	return EncodeResponse(rw, http.StatusCreated, resp)
+}
+
+func (s *Server) getAllPreheatTasks(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	tasks, err := s.PreheatMgr.GetAll(ctx)
+	if err != nil {
+		return err
+	}
+	return EncodeResponse(rw, http.StatusOK, tasks)
+}
+
+func (s *Server) getPreheatTask(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+	id := mux.Vars(req)["id"]
+	task, err := s.PreheatMgr.Get(ctx, id)
+	if err != nil {
+		return err
```

Comment:
You defined a code of StatusNotFound, while I am wondering if the code here could return the code of 404?

Reply:
`PreheatMgr` returns the error with a proper code and method`handlePreheatErrorResponse` convert the error to error response and send to user.